pybop/_problem.py

```diff
@@ -1,30 +1,82 @@
 import numpy as np
 
 
-class Problem:
+class BaseProblem:
     """
     Defines a PyBOP single output problem, follows the PINTS interface.
     """
 
     def __init__(
         self,
-        model,
         parameters,
-        dataset,
-        signal="Terminal voltage [V]",
+        model=None,
         check_model=True,
         init_soc=None,
         x0=None,
     ):
         self._model = model
+        if model is not None:
```

Comment:
I'm not sure this needs to be within a logic statement. If model = None, this still be calculated without issue, right?